contracts/modules/capital/Pool.sol

```diff
@@ -0,0 +1,307 @@
+      assets.push(_assets[i]);
+      minAmount[_assets[i]] = _minAmounts[i];
+      maxAmount[_assets[i]] = _maxAmounts[i];
+    }
+
+    master = INXMMaster(_master);
+    router = IUniswapV2Router01(_router);
+    twapOracle = TwapOracle(_twapOracle);
+    swapController = _swapController;
+  }
+
+  // fallback function
+  function() external payable {}
+
+  // for Pool1 upgrade compatibility
+  function sendEther() external payable {}
+
+  /* asset related functions */
+
+  function getAssets() external view returns (address[] memory) {
+    return assets;
+  }
+
+  function getAssetMinMax(address _asset) external view returns (uint min, uint max) {
+    return (minAmount[_asset], maxAmount[_asset]);
+  }
+
+  function addAsset(address _asset, uint _min, uint _max) external onlyGovernance {
+
+    for (uint i = 0; i < assets.length; i++) {
+      require(_asset != assets[i], 'asset exists');
+    }
+
+    assets.push(_asset);
+    minAmount[_asset] = _min;
+    maxAmount[_asset] = _max;
+  }
+
+  function removeAsset(address _asset) external onlyGovernance {
+
+    IERC20 token = IERC20(_asset);
+    uint tokenBalance = token.balanceOf(address(this));
+
+    require(tokenBalance == 0, 'balance must be 0');
+
+    for (uint i = 0; i < assets.length; i++) {
+
+      if (_asset != assets[i]) {
+        continue;
+      }
+
+      uint lastAssetIndex = assets.length - 1;
+      assets[i] = assets[lastAssetIndex];
+      assets.pop();
+
+      minAmount[_asset] = 0;
+      maxAmount[_asset] = 0;
+
+      return;
+    }
+
+    require(false, 'asset not found');
+  }
+
+  function setAssetMinMax(address _asset, uint _min, uint _max) external onlyGovernance {
+    require(_min <= _max, 'min > max');
+    minAmount[_asset] = _min;
+    maxAmount[_asset] = _max;
+  }
+
+  /* swap functions */
+
+  function getSwapQuote(
+    uint tokenAmountIn,
+    IERC20 fromToken,
+    IERC20 toToken
+  ) public view returns (uint tokenAmountOut) {
+
+    address[] memory path = new address[](2);
+    path[0] = address(fromToken);
+    path[1] = address(toToken);
+    uint[] memory amountsOut = router.getAmountsOut(tokenAmountIn, path);
+
+    return amountsOut[1];
+  }
+
+  function swapETHForTokens(
+    address toTokenAddress,
+    uint amountIn,
+    uint amountOutMin
+  ) external onlySwapController nonReentrant {
+
+    IERC20 toToken = IERC20(toTokenAddress);
+    uint balanceBefore = toToken.balanceOf(address(this));
+
+    // TODO: should we allow more frequent trades?
+    uint timeSinceLastTrade = block.timestamp.sub(lastSwapTime[toTokenAddress]);
+    require(timeSinceLastTrade > twapOracle.periodSize(), 'too fast');
+
+    {
+      // scope for liquidity check
+      address pairAddress = twapOracle.pairFor(ETH, toTokenAddress);
+      IUniswapV2Pair pair = IUniswapV2Pair(pairAddress);
+      (uint112 reserve0, uint112 reserve1, /* time */) = pair.getReserves();
+
+      uint ethReserve = ETH < toTokenAddress ? reserve0 : reserve1;
```

Comment:
could use a comment explanation here that reserves are sorted by token address in response